File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/c2/chaincode/api/fabric/config/HasSigncerts.java
package io.komune.c2.chaincode.api.fabric.config;

import com.google.common.base.Strings;
import io.komune.c2.chaincode.api.fabric.utils.FileUtils;

import java.net.MalformedURLException;
import java.net.URL;

public interface HasSigncerts {

    String getSigncerts();

    default URL getSigncertsAsUrl(String cryptoBase) throws MalformedURLException {
        return FileUtils.getUrl(cryptoBase, getSigncerts());
    }

}

File: c2-chaincode/chaincode-api/chaincode-api-config/src/main/kotlin/io/komune/c2/chaincode/api/config/utils/HasSigncerts.kt
package io.komune.c2.chaincode.api.config.utils

import java.net.MalformedURLException
import java.net.URL

interface HasSigncerts {
    val signcerts: String

    @Throws(MalformedURLException::class)
    fun getSigncertsAsUrl(cryptoBase: String): URL {
        return FileUtils.getUrl(cryptoBase, signcerts)
    }
}
